Objects/moduleobject.c

```diff
@@ -784,11 +786,62 @@ _PyModuleSpec_IsUninitializedSubmodule(PyObject *spec, PyObject *name)
     return rc;
 }
 
+static int
+_is_module_possibly_shadowing(PyObject *origin)
+{
+    // origin must be a unicode subtype
+    // Returns 1 if the module at origin could be shadowing a module of the
+    // same name later in the module search path. The condition we check is basically:
+    // not sys.flags.safe_path and os.path.dirname(origin) == (sys.path[0] or os.getcwd())
+    // Returns 0 otherwise (or if we aren't sure)
+    // Returns -1 if an error occurred that should be propagated
+    if (origin == NULL) {
+        return 0;
+    }
+    const PyConfig *config = _Py_GetConfig();
+    if (config->safe_path) {
+        return 0;
+    }
```

Comment:
Thanks, I made the change and it passes tests.

But when double checking it looks like it might be possible for `config->sys_path_0` to be non-null when `config->safe_path` is false if there's a `main_importer_path`?
https://github.com/python/cpython/blob/90a1b2859f99a4b07da6c46b99759444e3cefbfa/Modules/main.c#L581

Reply:
Okay, looks like this wasn't safe to make, e.g. `test.test_zipimport_support ` will hit the assert